src/service/onesignal/index.ts

```diff
@@ -14,7 +14,7 @@ const SET_TAG_TIMEOUT = 10 * 1000;
 const rebuildAndSubscribeTags = async () => {
   try {
     const response = await getSubscribeableTopic();
-    const {topics = [], history = []} = response?.data || {};
+    const {topics = [], history = [], hasUserPosted} = response?.data || {};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The destructuring of `hasUserPosted` from the response data is a good addition. However, it would be better to provide a default value for `hasUserPosted` in case it's not present in the response data. This will prevent potential issues where `hasUserPosted` might be `undefined`.

```diff
- const {topics = [], history = [], hasUserPosted} = response?.data || {};
+ const {topics = [], history = [], hasUserPosted = false} = response?.data || {};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->